lib/cn/csched_sp3_work.c

```diff
@@ -287,16 +321,52 @@ sp3_work_wtype_split(
     enum cn_rule             *rule)
 {
     struct cn_tree_node *tn = spn2tn(spn);
+    struct cn_tree *tree = tn->tn_tree;
     struct kvset_list_entry *le;
     struct list_head *head;
+    uint64_t now;
+    uint kvsets;
 
     head = &tn->tn_kvset_list;
     *mark = list_last_entry_or_null(head, typeof(*le), le_link);
+    kvsets = cn_ns_kvsets(&tn->tn_ns);
 
     *action = CN_ACTION_SPLIT;
     *rule = CN_RULE_SPLIT;
 
-    return cn_ns_kvsets(&tn->tn_ns);
+    mutex_lock(&tn->tn_ss_lock);
+    now = get_time_ns();
+
+    if (!tn->tn_ss_splitting) {
```

Comment:
What would be the scenario where tn_ss_splitting is true here? Doesn't the compaction token on this node protect against it?

Reply:
Never mind, I see it now after reading the last paragraph in your comment.